src/Rules/Rule.cs

```diff
@@ -20,7 +20,10 @@ abstract class Rule : IRule
         protected Rule()
         {
             // keep these two calls in the following order or the RuleInfo.Condition string won't get populated
+#pragma warning disable CA2214
             this.Condition = CreateCondition();
```

Comment:
Based on the docs for CA2214, it seems like the current code isn't guaranteed to work in the future. We may want to consider using dependency injection in the future, and having the ctor take a Condition object (built in the derived class) as a parameter.

Reply:
Agreed. That is a big change though. And I was trying to keep this PR limited in scope, quite obviously ineffectually.